### PR TITLE
2.x consistent get interactions

### LINKED ISSUE #1272: [AUTOCUT] Integration Test failed for ml-commons: 2.10.0 tar distribution
The integration test failed at distribution level for component ml-commons<br>Version: 2.10.0<br>Distribution: tar<br>Architecture: x64<br>Platform: linux<br><br>Please check the logs: https://build.ci.opensearch.org/job/integ-test/5500/display/redirect<br><br> * Test-report manifest:*<br> - https://ci.opensearch.org/ci/dbc/integ-test/2.10.0/8296/linux/x64/tar/test-results/5500/integ-test/test-report.yml <br><br> _Note: Steps to reproduce, additional logs and other files can be found within the above test-report manifest._

### PR DESCRIPTION
**Pull request by HenryL27**
### Description
Fix issue where getInteractions returned different results in security-enabled and -disabled settings. Also fix a data race in deleteConversation that made that test flaky.
 
### Issues Resolved
#1272 
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: memory/src/main/java/org/opensearch/ml/memory/index/ConversationMetaIndex.java [hunk 1]
```diff
@@ -290,7 +290,14 @@ public void checkAccess(String conversationId, ActionListener<Boolean> listener)
...
             }, e -> { internalListener.onFailure(e); });
-            client.get(getRequest, al);
+            client
+                .admin()
+                .indices()
+                .refresh(Requests.refreshRequest(indexName), ActionListener.wrap(refreshResponse -> { client.get(getRequest, al); }, e -> {
+                    log.error("Failed to refresh conversations index during check access ", e);
+                    internalListener.onFailure(e);
+                }));
+            // client.get(getRequest, al);
```
Comment (dhrubo-os) [inline memory/src/main/java/org/opensearch/ml/memory/index/ConversationMetaIndex.java:300]: Do we need this?

File: memory/src/main/java/org/opensearch/ml/memory/index/OpenSearchConversationalMemoryHandler.java [hunk 2]
```diff
@@ -247,19 +250,25 @@ public ActionFuture<List<ConversationMeta>> getConversations(int maxResults) {
...
-                    interactionsListener
-                        .whenComplete(interactionResult -> { listener.onResponse(metaResult && interactionResult); }, listener::onFailure);
+                interactionsListener
+                    .whenComplete(
+                        interactionResult -> { conversationMetaIndex.deleteConversation(conversationId, metaDeleteListener); },
+                        listener::onFailure
+                    );
+
+                metaDeleteListener.whenComplete(metaDeleteResult -> {
+                    log.info("SUCCESSFUL DELETION OF CID " + conversationId);
```
Comment (dhrubo-os) [inline memory/src/main/java/org/opensearch/ml/memory/index/OpenSearchConversationalMemoryHandler.java:268]: CID--> `Conversation Id: `

**Comment by HenryL27** [comment]
https://github.com/opensearch-project/opensearch-build/issues/3743

File: memory/src/main/java/org/opensearch/ml/memory/index/ConversationMetaIndex.java (hunk 1 above)
Comment (ylwu-amzn) [inline memory/src/main/java/org/opensearch/ml/memory/index/ConversationMetaIndex.java:300]: Should we remove this line?

File: memory/src/main/java/org/opensearch/ml/memory/index/OpenSearchConversationalMemoryHandler.java [hunk 3]
```diff
@@ -247,19 +250,25 @@ public ActionFuture<List<ConversationMeta>> getConversations(int maxResults) {
     public void deleteConversation(String conversationId, ActionListener<Boolean> listener) {
         StepListener<Boolean> accessListener = new StepListener<>();
         conversationMetaIndex.checkAccess(conversationId, accessListener);
-
+        log.info("DELETING CID " + conversationId);
```
Comment (ylwu-amzn) [inline memory/src/main/java/org/opensearch/ml/memory/index/OpenSearchConversationalMemoryHandler.java:253]: CID is hard to understand if someone who not familiar with this feature look into the log. How about change to "conversation"

File: memory/src/main/java/org/opensearch/ml/memory/index/ConversationMetaIndex.java (hunk 1 above)
Comment (HenryL27) [reply memory/src/main/java/org/opensearch/ml/memory/index/ConversationMetaIndex.java:300]: ack

File: memory/src/main/java/org/opensearch/ml/memory/index/OpenSearchConversationalMemoryHandler.java (hunk 3 above)
Comment (HenryL27) [reply memory/src/main/java/org/opensearch/ml/memory/index/OpenSearchConversationalMemoryHandler.java:253]: ack

[the remark this answers is not in the document]
Comment (HenryL27) [reply memory/src/main/java/org/opensearch/ml/memory/index/OpenSearchConversationalMemoryHandler.java:253]: hah yaliang had the same comments

**Review by dhrubo-os — APPROVED** [review]

**Review by ylwu-amzn — APPROVED** [review]

**Pull request merged**